### ISSUE TITLE
Exception: Current loss scale already at minimum - cannot decrease scale anymore

### ISSUE THREAD

**Issue by Z-eloto** [microsoft/LMOps#280]
First of all, thank you for sharing your codes. When running gpt2/kd/kd_medium.sh on 2*3090, the program encountered this error. What should I do, such as adjusting the learning rate?

**Comment by shiboyu1999** [comment]
You can use fp32 to train the model or decrease the batch size to 1.

**Comment by Z-eloto** [comment]
> You can use fp32 to train the model or decrease the batch size to 1.

Thanks. I will try it. :)

**Comment by t1101675** [comment]
You can also try using bfloat16 by replacing `ds_config_zero1_fp16.json` in this [line](https://github.com/microsoft/LMOps/blob/5697efccbe2d9c122718290025e92ef2392785eb/minillm/scripts/gpt2/kd/kd_medium.sh#L78) with `ds_config_zero1_bf16.json`.

**Comment by Z-eloto** [comment]
> You can also try using bfloat16 by replacing `ds_config_zero1_fp16.json` in this [line](https://github.com/microsoft/LMOps/blob/5697efccbe2d9c122718290025e92ef2392785eb/minillm/scripts/gpt2/kd/kd_medium.sh#L78) with `ds_config_zero1_bf16.json`.

OK, I have already resolved this problem. Thanks :)

**Comment by t1101675** [comment]
This will not affect the results much. In fact, bf16 will be more stable in training than fp16 and will not suffer from the "Current loss scale already at minimum" problem.

**Comment by Z-eloto** [comment]
I see. Many thanks!

> This will not affect the results much. In fact, bf16 will be more stable in training than fp16 and will not suffer from the "Current loss scale already at minimum" problem.
